pkg/nfd-worker/nfd-worker.go

```diff
@@ -190,11 +190,19 @@ func (i *infiniteTicker) Reset(d time.Duration) {
 
 // Run feature discovery.
 func (w *nfdWorker) runFeatureDiscovery() error {
+	discoveryStart := time.Now()
 	for _, s := range w.featureSources {
-		klog.V(2).Infof("running discovery for %q source", s.Name())
+		currentSourceStart := time.Now()
 		if err := s.Discover(); err != nil {
 			klog.Errorf("feature discovery of %q source failed: %v", s.Name(), err)
 		}
+		klog.V(2).Infof("discovery duration for %q source: %v", s.Name(), time.Since(currentSourceStart))
+	}
+
+	discoveryDuration := time.Since(discoveryStart)
+	klog.V(2).Infof("feature discovery for sources lasted for: %v", discoveryDuration)
```

Comment:
For improved log output amount control, per-source logging could be V(3), and this V(2), with count included to it:
```suggestion
	klog.V(2).Infof("feature discovery for %d sources lasted for: %v", len(w.featureSources), discoveryDuration)
```
(Just a suggestion, feel free to ignore.)

Reply:
@VillePihlava please address, I agree with @eero-t

Reply:
@eero-t PTAL

Reply:
This has now been changed

Reply:
@eero-t PTAL

Reply:
Looks fine now!